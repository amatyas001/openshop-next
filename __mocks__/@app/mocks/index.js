/**
 * Generates mock products with unique attributes.
 *
 * @param {*} count
 * @returns List of mock items with dynamic values
 */
export const mockProduct = (count = 1) => {
  let index = 0;
  let items = [];
  do {
    items.push({
      id: `mock_id_${index}`,
      amount: 1,
      color: `mock_color_${index}`,
      description: `mock_description_${index}`,
      gender: `mock_gender_${index}`,
      img: `mock_img_${index}`,
      name: `mock_name_${index}`,
      price: 1,
      review: `mock_review_${index}`,
      starrating: 1,
    });
    index++;
  } while (--count !== 0);
  return items;
};

/**
 * Generates mock products with buying amount details.
 *
 * @param {*} count Number of products to generate
 * @param {*} amount Mock amount attached to the listed products
 * @returns List of mock products with mock buying amount details
 */
export const mockProductWithBuyAmount = (count, amount = 1) => {
  return mockProduct(count).map((c) => ({
    ...c,
    buy: { amount },
  }));
};

/**
 * Generates mock products with unique price attributes.
 *
 * @param {*} count Number of products to generate
 * @param {*} value Base value of prices attached to generated products
 * @returns List of mock products with dynamic values and prices
 */
export const mockProductWithPrice = (count, value) => {
  const generatePrices = function* (base) {
    let index = 1;
    while (true) yield base * index++;
  };
  const generator = generatePrices(value);
  return mockProduct(count).map((c) => ({
    ...c,
    price: generator.next().value,
  }));
};

/**
 * Generates mock products with unique color attributes.
 * Each odd products has an array format color property.
 *
 * @param {number} count Number of products to generate
 * @returns Object with exact shape of
 * - `colors`: List of generated unique colors.
 * - `products`: List of mock items with dynamic values and colors.
 */
export const mockProductWithColor = (count) => {
  const generateColors = function* () {
    let index = 0;
    while (true) {
      yield index % 2 ? `mock_color_${index}` : [`mock_color_${index}`, `mock_color_${index + 1}`];
      index++;
    }
  };
  const generator = generateColors();
  const products = mockProduct(count).map((c, i) => ({
    ...c,
    color: generator.next().value,
  }));
  return {
    colors: [...new Set(products.reduce((a, c) => a.concat(c.color), ['All']))],
    products,
  };
};
