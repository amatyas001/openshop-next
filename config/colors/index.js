export * from './_button';
export * from './_panel';
export * from './_general';
