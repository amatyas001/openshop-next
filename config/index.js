export * from './colors';
export * from './menus/footer';
export * from './menus/main';
export * from './fields/payment';
export * from './state/initial';
