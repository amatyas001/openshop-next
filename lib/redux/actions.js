export * from './cart';
export * from './filter';
export * from './panel';
export * from './payment';
