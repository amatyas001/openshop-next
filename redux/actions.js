export const ADD_TO_CART = 'ADD_TO_CART';
export const REMOVE_FROM_CART = 'REMOVE_FROM_CART';
export const CLEAR_CART = 'CLEAR_CART';
export const GET_AMOUNT = 'GET_AMOUNT';
export const CHECKOUT_PROGRESS = 'CHECKOUT_PROGRESS';
export const PAYMENT_PROGRESS = 'PAYMENT_PROGRESS';
export const PAYMENT_RESET = 'PAYMENT_RESET';

/**
 *  CART ACTIONS
 */
export function addToCart(item) {
  return {
    type: ADD_TO_CART,
    payload: item,
  };
}

export function removeFromCart(item) {
  return {
    type: REMOVE_FROM_CART,
    payload: item,
  };
}

export function clearCart() {
  return {
    type: CLEAR_CART,
  };
}

export function getAmount() {
  return {
    type: GET_AMOUNT,
  };
}

/**
 *  CHECKOUT FLOW
 */
export function checkoutReview() {
  return {
    type: CHECKOUT_PROGRESS,
    payload: { status: 'review' },
  };
}

export function checkoutDetails() {
  return {
    type: CHECKOUT_PROGRESS,
    payload: { status: 'details' },
  };
}

export function checkoutFinished() {
  return {
    type: CHECKOUT_PROGRESS,
    payload: { status: 'finished' },
  };
}

export function paymentActive() {
  return {
    type: PAYMENT_PROGRESS,
    payload: { status: true },
  };
}

export function paymentStart() {
  return {
    type: PAYMENT_PROGRESS,
    payload: { status: 'loading' },
  };
}

/**
 *  PAYMENT FLOW
 */
export function paymentConfirm(card, details, intent) {
  return {
    type: PAYMENT_PROGRESS,
    payload: {
      status: 'confirm',
      card,
      details,
      intent,
    },
  };
}

export function paymentSuccess(intent) {
  return {
    type: PAYMENT_PROGRESS,
    payload: {
      status: 'success',
      intent,
    },
  };
}

export function paymentCanceled(intent) {
  return {
    type: PAYMENT_PROGRESS,
    payload: {
      status: 'canceled',
      intent,
    },
  };
}

export function paymentError(error) {
  return {
    type: PAYMENT_PROGRESS,
    payload: { status: 'error', error },
  };
}

export function paymentReset() {
  return {
    type: PAYMENT_RESET,
  };
}

export function cancelPayment(id) {
  return async (dispatch) => {
    try {
      dispatch(paymentStart());
      const { loadStripe } = await import('@stripe/stripe-js');
      const stripe = await loadStripe(
        process.env.NEXT_PUBLIC_STRIPE_PUBLIC_KEY
      );
      stripe.paymentIntents.cancel(id, (err, intent) => {
        if (err) dispatch(paymentError(e.message));
        else dispatch(paymentCanceled(intent.id));
      });
    } catch (e) {
      dispatch(paymentError(e.message));
    }
  };
}

export function intentCart(card, details, items) {
  return async (dispatch) => {
    try {
      dispatch(paymentStart());
      const axios = await import('axios');
      const intent = await axios.post(
        process.env.NODE_ENV === 'production'
          ? 'https://openshop.netlify.com/.netlify/functions/intent'
          : 'http://localhost:9000/.netlify/functions/intent',
        {
          items,
        },
        {
          headers: { 'Content-Type': 'application/json' },
        }
      );
      dispatch(paymentConfirm(card, details, intent.data));
    } catch (e) {
      dispatch(paymentError(e.message));
    }
  };
}
