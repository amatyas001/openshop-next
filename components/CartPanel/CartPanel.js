import Link from 'next/link';
import { useSelector, useDispatch } from 'react-redux';
import { paymentReview } from '../../redux/actions';
import { Collapse, Flex, Heading, Text, SimpleGrid } from '@chakra-ui/core';
import { Button, CartContent } from '../index';
import { MdShoppingCart } from 'react-icons/md';

export const CartPanel = (props) => {
  const [show, setShow] = React.useState(false);
  const { cart } = useSelector((state) => state);
  const dispatch = useDispatch();

  return (
    <Flex
      as={'section'}
      role='complementary'
      id='cart'
      flexDirection='column'
      {...props}
    >
      {/* cart button */}
      <Button
        as={'a'}
        role='button'
        d='inline-block'
        width='30px'
        ml='auto'
        aria-controls='cart-holder'
        backgroundColor='transparent'
        color='gray.800'
        border='0'
        onClick={() => setShow(!show)}
        id='card-toggler'
        _hover={{
          bg: '',
          color: 'purple.400',
        }}
        _active={{
          color: 'purple.200',
        }}
      >
        <Text mb='-4px' fontSize='1.7rem' as={MdShoppingCart} />
      </Button>

      {cart && (
        <>
          {/* item count icon */}
          <Text
            as='span'
            d={cart.length ? 'block' : 'none'}
            w='20px'
            h='20px'
            position='absolute'
            top='5px'
            right='10px'
            bg='purple.600'
            borderRadius='50%'
            textAlign='center'
            fontSize='1rem'
            color='gray.100'
            fontWeight='bold'
          >
            {cart.length}
          </Text>

          {/* panel */}
          <Collapse
            isOpen={show}
            position='fixed'
            top='60px'
            right='0'
            width={{ sm: '100%', md: '70%', lg: '60%', xl: '40%' }}
            bg='white'
            border='1px'
            borderTop='0'
            borderColor='purple.400'
            boxShadow='2px 7px 7px black'
            px='10px'
            py='5px'
          >
            {/* items */}
            {cart.length ? (
              <CartContent />
            ) : (
              <Heading textAlign='center'>no items in your cart</Heading>
            )}

            <SimpleGrid columns='2' spacing='15px'>
              {/* close button */}
              <Button
                onClick={() => setShow(false)}
                bg='gray.200'
                color='purple.600'
                width='100%'
                mr='15px'
              >
                close
              </Button>

              {/* checkout button */}
              <Link href='/checkout' passHref>
                <Button
                  onClick={() => {
                    setShow(false);
                    dispatch(paymentReview());
                  }}
                  bg='purple.800'
                  color='gray.200'
                  width='100%'
                >
                  continue
                </Button>
              </Link>
            </SimpleGrid>
          </Collapse>
        </>
      )}
    </Flex>
  );
};
