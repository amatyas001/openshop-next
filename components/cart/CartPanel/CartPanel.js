import PropTypes from 'prop-types';
import Link from 'next/link';
import { useSelector, useDispatch } from 'react-redux';
import { MdShoppingCart } from 'react-icons/md';
import {
  Box,
  Collapse,
  Flex,
  Heading,
  Text,
  SimpleGrid,
} from '@chakra-ui/core';
import { paymentReview } from '@app/redux/actions';
import { Button, CartContent } from '@app/components';

/**
 * Wrapper component for the cart. Displays the cart button
 * and a fixed position collapse menu containing the cart content.
 *
 * When the cart is empty it renders *"no items in your cart"* text.
 *
 * At the bottom, the component displays the cart control buttons
 * used to close the panel and initiate payment process.
 *
 * ***State Dependencies***
 * - `cart`
 *
 * ***Wrapped Components***
 * - [CartContent](#cartcontent)
 *
 * @example
 * ```jsx
 * <CartPanel
 *  panel={stylePropsObject}
 *  icon={<CustomIcon />}
 *  indicator={<CustomIndicator />}
 *  close={customStyles}
 *  continue={customStlyes}
 * />
 * ```
 */
export const CartPanel = (props) => {
  const [show, setShow] = React.useState(false);
  const { cart = [] } = useSelector((state) => state);
  const dispatch = useDispatch();

  return (
    <Flex
      as={'section'}
      role='complementary'
      id='cart'
      flexDirection='column'
      {...props}
    >
      {/* icon button */}
      <Button
        as={'a'}
        role='button'
        d='inline-block'
        width='30px'
        ml='auto'
        aria-label='cart-toggler'
        backgroundColor='transparent'
        color='gray.800'
        border='0'
        onClick={() => setShow(!show)}
        id='cart-toggler'
        _hover={{
          bg: '',
          color: 'purple.400',
        }}
        _active={{
          color: 'purple.200',
        }}
      >
        {props.icon || <Box mb='-4px' fontSize='1.7rem' as={MdShoppingCart} />}
      </Button>
      {/* indicator */}
      <Text
        as='span'
        d={cart.length ? 'block' : 'none'}
        w='20px'
        h='20px'
        position='absolute'
        top='5px'
        right='10px'
        bg='purple.600'
        borderRadius='50%'
        textAlign='center'
        fontSize='1rem'
        color='gray.100'
        fontWeight='bold'
        {...props.indicator}
      >
        {cart.length}
      </Text>

      {/* panel */}
      <Collapse
        isOpen={show}
        position='fixed'
        top='60px'
        right='0'
        width={{
          sm: '100%',
          md: '70%',
          lg: '60%',
          xl: '40%',
        }}
        bg='white'
        border='1px'
        borderTop='0'
        borderColor='purple.400'
        boxShadow='2px 7px 7px black'
        px='2%'
        py='5px'
        {...props.panel}
      >
        {/* items */}
        {cart.length ? (
          <CartContent />
        ) : (
          <Heading textAlign='center'>no items in your cart</Heading>
        )}

        <SimpleGrid columns='2' spacing='15px'>
          {/* close button */}
          <Button
            onClick={() => setShow(false)}
            bg='gray.200'
            color='purple.600'
            width='100%'
            mr='15px'
            {...props.close}
          >
            close
          </Button>

          {/* checkout button */}
          <Link href='/checkout' passHref>
            <Button
              onClick={() => {
                setShow(false);
                dispatch(paymentReview());
              }}
              bg='purple.800'
              color='gray.200'
              width='100%'
              {...props.continue}
            >
              continue
            </Button>
          </Link>
        </SimpleGrid>
      </Collapse>
    </Flex>
  );
};

CartPanel.defaultProps = {
  panel: null,
  indicator: null,
  icon: null,
  close: null,
  continue: null,
};

CartPanel.propTypes = {
  /**
   * [Style Props](https://chakra-ui.com/style-props) for collapsable panel
   */
  panel: PropTypes.object,

  /**
   * [Style Props](https://chakra-ui.com/style-props) for indicator
   */
  indicator: PropTypes.object,

  /**
   * Custom element for cart icon
   */
  icon: PropTypes.element,

  /**
   * [Style Props](https://chakra-ui.com/style-props) for close button
   */
  close: PropTypes.element,

  /**
   * [Style Props](https://chakra-ui.com/style-props) for continue button
   */
  continue: PropTypes.object,
};
