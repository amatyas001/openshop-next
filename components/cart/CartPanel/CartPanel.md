```jsx
import { useDispatch, useSelector } from 'react-redux';
import { addToCart } from '@app/redux/actions';
import { Button } from '@app/components';

const dispatch = useDispatch();
const { cart } = useSelector((store) => store);

const items = [
  {
    id: 0,
    img: '1.jpg',
    name: 'My fancy item',
    desc: 'The best item in the world',
    color: 'Rainbow',
    price: 100,
  },
  {
    id: 1,
    img: '2.jpg',
    name: 'My cool item',
    desc: 'The best item in the world',
    color: 'Pink',
    price: 30,
  },
  {
    id: 2,
    img: '3.jpg',
    name: 'My awesome item',
    desc: 'The best item in the world',
    color: 'Brown',
    price: 150,
  },
];

const index = (cart && cart.length) || 0;

<>
  <CartPanel panel={{ width: '95.8%', top: '120px' }} mr='-40px' mt='-5px' />
  <Button
    onClick={() => {
      index !== items.length && dispatch(addToCart(items[index]));
    }}
  >
    Click here to add items to cart ({items.length - index} remaining)
  </Button>
</>;
```
