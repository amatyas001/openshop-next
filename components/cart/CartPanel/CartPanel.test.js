import { create, act } from 'react-test-renderer';
import { CartPanel } from './CartPanel';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { INITIAL_STATE } from '@app/config/';
import { mockProduct } from '@app/mocks';
import { panelToggle, paymentReview } from '@app/lib/redux/actions';

const mockStore = configureStore([]);
jest.mock('next/link', () => ({ children }) => children);

describe('<CartPanel />', () => {
  let tree, store;

  describe('on empty cart', () => {
    beforeAll(() => {
      act(() => {
        store = mockStore(INITIAL_STATE);

        tree = create(
          <Provider store={store}>
            <CartPanel />
          </Provider>
        );
      });
    });

    it('should render without props', () => {
      expect(tree.toJSON()).toMatchSnapshot();
    });

    it('should close panel with button', () => {
      act(() => {
        tree.root
          .findByProps({
            'data-testid': 'cart-panel-button-close',
          })
          .props.onClick();
      });
      expect(store.getActions()).toEqual([panelToggle('cart', false)]);
    });

    it('should render disabled continue button', () => {
      expect(
        tree.root.findByProps({
          'data-testid': 'cart-panel-button-continue',
        }).props.disabled
      ).toBeTruthy();
    });
  });

  describe('on non empty cart', () => {
    beforeAll(() => {
      store = mockStore({
        ...INITIAL_STATE,
        cart: [{ ...mockProduct(1)[0], buy: { amount: 1 } }],
      });

      act(() => {
        tree = create(
          <Provider store={store}>
            <CartPanel />
          </Provider>
        );
      });
    });

    it('should render without props', () => {
      expect(tree.toJSON()).toMatchSnapshot();
    });

    it('should render enabled continue button', () => {
      expect(
        tree.root.findByProps({
          'data-testid': 'cart-panel-button-continue',
        }).props.disabled
      ).toBeFalsy();
    });

    it('should continue to payment review', () => {
      act(() => {
        tree.root
          .findByProps({
            'data-testid': 'cart-panel-button-continue',
          })
          .props.onClick({
            preventDefault: jest.fn(),
            currentTarget: {
              nodeName: 'button',
            },
          });
      });

      expect(store.getActions()).toEqual([
        panelToggle('cart', false),
        paymentReview(),
      ]);
    });
  });
});
