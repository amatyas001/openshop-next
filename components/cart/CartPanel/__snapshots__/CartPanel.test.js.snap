// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartPanel /> on empty cart should render without props 1`] = `
<section
  className="css-1ofqig9"
  id="cart"
  role="complementary"
>
  <button
    aria-label="cart-toggler"
    className="css-1vus3us"
    data-testid="cart-panel-button"
    href="#void"
    id="cart-toggler"
    onClick={[Function]}
    onKeyDown={[Function]}
    tabIndex="0"
    type="button"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
          "marginBottom": "-5px",
          "marginRight": "5px",
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zM1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2H1zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
      />
    </svg>
    0 item(s)
  </button>
  <div
    aria-hidden={true}
    className="rah-static rah-static--height-zero css-vppuuz"
    style={
      Object {
        "height": 0,
        "overflow": "hidden",
      }
    }
  >
    <div
      style={
        Object {
          "WebkitTransition": "opacity 250ms ease 0ms",
          "opacity": 0,
          "transition": "opacity 250ms ease 0ms",
        }
      }
    >
      <div
        className="css-rl10s"
      >
        <div
          className="css-1fm8xe3"
        >
          <button
            className="css-1ks4ucs"
            data-testid="cart-panel-button-close"
            href="#void"
            onClick={[Function]}
            onKeyDown={[Function]}
            tabIndex="0"
            type="button"
          >
            close
          </button>
          <button
            className="css-48tyil"
            data-testid="cart-panel-button-continue"
            disabled={true}
            href="#void"
            onClick={[Function]}
            onKeyDown={[Function]}
            tabIndex="0"
            type="button"
          >
            continue
          </button>
          <div
            className="css-ufscxh"
            data-testid="cart-panel-amount"
          >
            <p
              className="css-dptndp"
            >
              TOTAL
            </p>
            <strong
              className="css-lk14dv"
              data-testid="cart-amount-value"
            >
              0.00
               $
            </strong>
          </div>
        </div>
        <ul
          className="css-1q2qbgw"
          data-testid="cart-panel-content"
          role="list"
        />
      </div>
    </div>
  </div>
</section>
`;

exports[`<CartPanel /> on non empty cart should render without props 1`] = `
<section
  className="css-1ofqig9"
  id="cart"
  role="complementary"
>
  <button
    aria-label="cart-toggler"
    className="css-1vus3us"
    data-testid="cart-panel-button"
    href="#void"
    id="cart-toggler"
    onClick={[Function]}
    onKeyDown={[Function]}
    tabIndex="0"
    type="button"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
          "marginBottom": "-5px",
          "marginRight": "5px",
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zM1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2H1zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2z"
      />
    </svg>
    1 item(s)
  </button>
  <div
    aria-hidden={true}
    className="rah-static rah-static--height-zero css-vppuuz"
    style={
      Object {
        "height": 0,
        "overflow": "hidden",
      }
    }
  >
    <div
      style={
        Object {
          "WebkitTransition": "opacity 250ms ease 0ms",
          "opacity": 0,
          "transition": "opacity 250ms ease 0ms",
        }
      }
    >
      <div
        className="css-rl10s"
      >
        <div
          className="css-1fm8xe3"
        >
          <button
            className="css-1ks4ucs"
            data-testid="cart-panel-button-close"
            href="#void"
            onClick={[Function]}
            onKeyDown={[Function]}
            tabIndex="0"
            type="button"
          >
            close
          </button>
          <button
            className="css-48tyil"
            data-testid="cart-panel-button-continue"
            disabled={false}
            href="#void"
            onClick={[Function]}
            onKeyDown={[Function]}
            tabIndex="0"
            type="button"
          >
            continue
          </button>
          <div
            className="css-ufscxh"
            data-testid="cart-panel-amount"
          >
            <p
              className="css-dptndp"
            >
              TOTAL
            </p>
            <strong
              className="css-lk14dv"
              data-testid="cart-amount-value"
            >
              1.00
               $
            </strong>
          </div>
        </div>
        <ul
          className="css-1q2qbgw"
          data-testid="cart-panel-content"
          role="list"
        >
          <li
            className="css-1tp0oa"
            role="listitem"
          >
            <img
              alt="mock_name_0"
              className="css-rhemcv"
            />
            <div
              className="css-zbq0po"
            >
              <strong
                className="css-airqvi"
              >
                mock_name_0
              </strong>
              <em
                className="css-1fji79j"
              >
                mock_color_0 in One Size x1
              </em>
            </div>
            <strong
              className="css-1som8bm"
            >
              1.00
               $
            </strong>
            <button
              className="css-5sp2m4"
              data-testid="cart-item-button-delete"
              href="#void"
              onClick={[Function]}
              onKeyDown={[Function]}
              tabIndex="0"
              type="button"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": undefined,
                  }
                }
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
                />
              </svg>
            </button>
          </li>
        </ul>
      </div>
    </div>
  </div>
</section>
`;
